Samples/Client/SwiftClient/SwiftClient/VideoStreamViewController.swift

```diff
@@ -0,0 +1,640 @@
+        
+        initGestureRecognizer()
+        connect()
+    }
+    
+    func joinPeer(peerId: Int) {
+        createPeerConnection(peerId: peerId)
+        inputChannel = self.peerConnection.dataChannel(forLabel: "inputDataChannel", configuration: RTCDataChannelConfiguration())
+        inputChannel.delegate = self
+        
+        let offerOptions = RTCMediaConstraints(mandatoryConstraints: ["OfferToReceiveAudio": "false","OfferToReceiveVideo": "true"], optionalConstraints: nil)
+        
+        peerConnection.offer(for: offerOptions) { (sdp, error) in
+            let sdpH264 = sdp!.sdp.replacingOccurrences(of: "96 98 100 102", with: "100 96 98 102")
+            let receivedOffer = RTCSessionDescription(type: .offer, sdp: sdpH264)
+            self.peerConnection.setLocalDescription(receivedOffer, completionHandler: { (error) in
+                
+                let jsonData = [
+                    "type": "offer",
+                    "sdp": receivedOffer.sdp
+                ]
+                
+                self.sendToPeer(peerId: peerId, data: self.convertDictionaryToData(dict: jsonData)!)
+            })
+        }
+    }
+    
+    func createPeerConnection(peerId: Int) {
+        newIceCandidatePeerId = peerId
+        
+        let defaultPeerConnectionConstraints = RTCMediaConstraints(mandatoryConstraints: nil,
+                                                                   optionalConstraints: ["DtlsSrtpKeyAgreement": "true"])
+        let rtcConfig = RTCConfiguration()
+        let iceServer = RTCIceServer(urlStrings: ["turn:turnserver3dstreaming.centralus.cloudapp.azure.com:5349"],
+                                     username: "user",
+                                     credential: "3Dtoolkit072017",
+                                     tlsCertPolicy: .insecureNoCheck)
+        rtcConfig.iceServers = [iceServer]
+        rtcConfig.iceTransportPolicy = .relay
+        
+        peerConnection = peerConnectionFactory.peerConnection(with: rtcConfig, constraints: defaultPeerConnectionConstraints, delegate: self)
+    }
+    
+    func handlePan(_ recognizer: UIPanGestureRecognizer) {
+        
+        print(recognizer.numberOfTouches)
+        
+        switch recognizer.state {
+        case .began:
+            navOnFingerDown(recognizer: recognizer)
+        case .changed:
+            navOnFingerMove(recognizer: recognizer)
+        case .ended:
+            navOnFingerUp()
+        default:
+            break
+        }
+    }
+    
+    func initGestureRecognizer() {
+        let panGestureRecognizer = UIPanGestureRecognizer(target: self, action: #selector(VideoStreamViewController.handlePan(_:)))
+        panGestureRecognizer.minimumNumberOfTouches = 1
+        panGestureRecognizer.maximumNumberOfTouches = 2
+        
+        self.renderView.addGestureRecognizer(panGestureRecognizer)
+    }
+    
+    func navOnFingerDown(recognizer: UIPanGestureRecognizer) {
+        
```

Comment:
nit: can we remove this space to try and be more consistent? seems in the rest of the codebase you generally don't leave a line before doing work

Reply:
👍